src/workerd/jsg/resource.h

```diff
@@ -191,6 +192,94 @@ template <typename TypeWrapper,
     typename Indexes>
 struct MethodCallback;
 
+// Helper to determine if a type is compatible with V8 fast API
+// TODO(soon): Add support for v8::Local<v8::Value>
+template <typename T>
+struct IsFastApiCompatibleType {
+  static constexpr bool value = std::is_same_v<T, void> || std::is_same_v<T, bool> ||
+      std::is_same_v<T, int32_t> || std::is_same_v<T, uint32_t> || std::is_same_v<T, float> ||
+      std::is_same_v<T, double> ||
+      std::is_pointer_v<T>;  // Allow pointers, which can be embedder types
+};
+
+// Helper to determine if a method can be used with V8 fast API
+template <typename Ret, typename... Args>
+struct IsFastApiCompatible {
+  static constexpr bool value =
+      IsFastApiCompatibleType<Ret>::value && (IsFastApiCompatibleType<Args>::value && ...);
+};
+
+// Helper to determine if a method pointer type is compatible with Fast API
+template <typename Method>
+struct IsFastMethodCompatible {
+  static constexpr bool value = false;
+};
+
+// Specialization for non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...)> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
+
+// Specialization for const non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...) const> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
```

Comment:
I think another key difference here that we need to be careful of is that the type mappings for fast api calls will *not* pass through our TypeWrapper mechanism. This is generally a good thing but might be surprising for folks. It will be worth documenting this.

Reply:
We can enable type mappings now, since all of the limitations are gone.